core/src/test/java/org/bitcoinj/script/ScriptTest.java

```diff
@@ -394,7 +395,17 @@ public void dataDrivenInvalidTransactions() throws Exception {
             if (test.isArray() && test.size() == 1 && test.get(0).isTextual())
                 continue; // This is a comment.
             Map<TransactionOutPoint, Script> scriptPubKeys = parseScriptPubKeys(test.get(0));
-            Transaction transaction = TESTNET.getDefaultSerializer().makeTransaction(HEX.decode(test.get(1).asText().toLowerCase()));
+            byte[] txBytes = HEX.decode(test.get(1).asText().toLowerCase());
```

Comment:
I wonder if perhaps some added testcases to `tx_invalid.json` are missing here? This testcase succeeds with or without any of the core logic changes.

Reply:
Without this modification, the test fails because the new parser cannot handle this zero-input, one-output transaction `0100000000010000000000000000015100000000`. It tries to parse it as segwit transaction and fails because it is incorrectly formed.

The old Transaction code was parsing it as a segwit tx without failing but the result was invalid: it was reading it as a zero-input and zero-output tx (the raw tx has one output). The actual test was passing because the point was to detect invalid transactions, not incorrectly parsed ones.

Reply:
Ok thanks for the explanation.